jbpm-bpmn2/src/main/java/org/jbpm/bpmn2/xml/AbstractNodeHandler.java

```diff
@@ -544,13 +545,28 @@ protected void writeVariableName(EventNode eventNode, StringBuilder xmlDump) {
 		}
 	}
 	
-    protected String getSignalExpression(NodeImpl node) {
+    protected String getSignalExpression(NodeImpl node, String signalName, String variable) {
         String signalExpression = RUNTIME_SIGNAL_EVENT;
         String scope = (String) node.getMetaData("customScope");
         if ("processInstance".equalsIgnoreCase(scope)) {
-            signalExpression = PROCESS_INSTANCE_SIGNAL_EVENT;
+            signalExpression = PROCESS_INSTANCE_SIGNAL_EVENT +  "org.jbpm.process.instance.impl.util.VariableUtil.resolveVariable(\""+ signalName + "\", kcontext.getNodeInstance()), " + (variable == null ? "null" : variable) + ");";
         } else if ("runtimeManager".equalsIgnoreCase(scope) || "project".equalsIgnoreCase(scope)) {
-            signalExpression = RUNTIME_MANAGER_SIGNAL_EVENT;
+            signalExpression = RUNTIME_MANAGER_SIGNAL_EVENT + "org.jbpm.process.instance.impl.util.VariableUtil.resolveVariable(\""+ signalName + "\", kcontext.getNodeInstance()), " + (variable == null ? "null" : variable) + ");";
+        } else if ("external".equalsIgnoreCase(scope)) {
+            signalExpression = "org.drools.core.process.instance.impl.WorkItemImpl workItem = new org.drools.core.process.instance.impl.WorkItemImpl();" + EOL +
+            "workItem.setName(\"External Send Task\");" + EOL + 
+            "workItem.setNodeInstanceId(kcontext.getNodeInstance().getId());" + EOL +
+            "workItem.setProcessInstanceId(kcontext.getProcessInstance().getId());" + EOL + 
+            "workItem.setNodeId(kcontext.getNodeInstance().getNodeId());" + EOL +
+            "workItem.setDeploymentId((String) kcontext.getKnowledgeRuntime().getEnvironment().get(\"deploymentId\"));" + EOL +
+            "workItem.setParameter(\"Signal\", org.jbpm.process.instance.impl.util.VariableUtil.resolveVariable(\""+ signalName + "\", kcontext.getNodeInstance()));" + EOL +
+            "workItem.setParameter(\"SignalProcessInstanceId\", kcontext.getVariable(\"SignalProcessInstanceId\"));" + EOL +
+            "workItem.setParameter(\"SignalWorkItemId\", kcontext.getVariable(\"SignalWorkItemId\"));" + EOL +
+            "workItem.setParameter(\"SignalDeploymentId\", kcontext.getVariable(\"SignalDeploymentId\"));" + EOL +
+            (variable == null ? "" : "workItem.setParameter(\"Data\", " + variable + ");" + EOL) +            
+            "((org.drools.core.process.instance.WorkItemManager) kcontext.getKnowledgeRuntime().getWorkItemManager()).internalExecuteWorkItem(workItem);";
+        } else {
+            signalExpression = signalExpression +  "org.jbpm.process.instance.impl.util.VariableUtil.resolveVariable(\""+ signalName + "\", kcontext.getNodeInstance()), " + (variable == null ? "null" : variable) + ");";
```

Comment:
Is it possible to add this logic to a class, and then have the script expression instantiate that class and execute a method? (A very little bit like of like a `WorkItemHandler`)

(Personal preference: I'd rather not have this amount of "constant" text)